### PR TITLE
Update to PathPlannerLib 2023

### PR DESCRIPTION
**Pull request by auscompgeek**

### CODE REVIEW

File: gen/PathPlannerTrajectory.yml [hunk 1]
```diff
@@ -5,31 +5,69 @@ classes:
...
+      getEndStopEvent:
       sample:
       getStates:
+        overloads:
+          '[const]':
+          "":
+      getMarkers:
+        overloads:
+          '[const]':
+          "":
```
Comment (auscompgeek) [inline gen/PathPlannerTrajectory.yml:27]: There's a bunch of const-qualified overloads on these methods now. The const overloads return copies, whereas the non-const overloads return references. We need to decide which is the most appropriate.

File: gen/PathPlannerTrajectory.yml [hunk 2]
```diff
@@ -5,31 +5,69 @@ classes:
...
+      position:
+  PathPlannerTrajectory::StopEvent:
+    attributes:
+      names:
+      executionBehavior:
+      waitBehavior:
+      waitTime:
+    enums:
+      ExecutionBehavior:
+      WaitBehavior:
```
Comment (auscompgeek) [inline gen/PathPlannerTrajectory.yml:73]: I'm seeing a bunch of compile errors trying to refer to `pathplanner::StopEvent::WaitBehavior::BEFORE` et al... should be under `pathplanner::PathPlannerTrajectory::StopEvent`.

File: gen/PathPlanner.yml [hunk 3]
```diff
@@ -7,5 +7,22 @@ classes:
...
+        overloads:
+          PathConstraints, bool, PathPoint, PathPoint, std::initializer_list<PathPoint>:
+            ignore: true
+          ? units::meters_per_second_t, units::meters_per_second_squared_t, bool,
+            PathPoint, PathPoint, std::initializer_list<PathPoint>
+          : ignore: true
+          PathConstraints, PathPoint, PathPoint, std::initializer_list<PathPoint>:
+            ignore: true
+          units::meters_per_second_t, units::meters_per_second_squared_t, PathPoint, PathPoint, std::initializer_list<PathPoint>:
+            ignore: true
```
Comment (auscompgeek) [inline gen/PathPlanner.yml:34]: Yes, in 2023.2.0 these really all take `initializer_list`. `vector` overloads should be available in the next release:

- https://github.com/mjansen4857/pathplanner/pull/178#issuecomment-1367091321

File: gen/PathConstraints.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  PathConstraints:
+    attributes:
+      maxVelocity:
+      maxAcceleration:
+    methods:
+      PathConstraints:
+        overloads:
+          units::meters_per_second_t, units::meters_per_second_squared_t:
+          "":
+            ignore: true
```
Comment (auscompgeek) [inline gen/PathConstraints.yml:13]: Java doesn't have the default constructor, hence the ignore.

File: gen/PathPlannerTrajectory.yml [hunk 5]
```diff
@@ -5,31 +5,69 @@ classes:
     methods:
       PathPlannerTrajectory:
         overloads:
-          std::vector<Waypoint>, units::meters_per_second_t, units::meters_per_second_squared_t, bool:
-          std::vector<PathPlannerState>:
+          std::vector<Waypoint>&, std::vector<EventMarker>&, PathConstraints, bool:
```
Comment (auscompgeek) [inline gen/PathPlannerTrajectory.yml:8]: `Waypoint` appears to be private now. This constructor overload is `protected` in the Java version.
```suggestion
          std::vector<Waypoint>&, std::vector<EventMarker>&, PathConstraints, bool:
          	ignore: true  # private Waypoint class, Java protected overload
```

File: gen/PathPlannerTrajectory.yml (hunk 1 above)
Comment (virtuald) [reply gen/PathPlannerTrajectory.yml:27]: If it's returning a vector, it's always getting copied regardless whether it's a reference or not.

Comment (auscompgeek) [reply gen/PathPlannerTrajectory.yml:27]: Good point. Still need to figure out whether we should take a copy or reference for the `get*State` methods though.

**Pull request merged**